### ISSUE TITLE
Problem with $by_correlation_id projection

### ISSUE THREAD

**Issue by krofka** [EventStore/EventStore#1875]
Hi,

I found the problem with $by_correlation_id in ES version 5, when using this library to append events through HTTP API (https://groups.google.com/forum/#!topic/event-store/1_e1yZfeWP0). The library sends json with escaped quotes in data and metadata, so the projection is not working.

Is it problem of the library or EventStore? The both formats are valid jsons I think.

Thanks,
Jakub Koci

**Comment by shaan1337** [comment]
Hello @krofka thanks for your report.

From the google group link, you mentioned that you tried the following JSON object:
```
[
  {
    "eventId": "someId",
    "eventType": "someType",
    "data": "{\"aggragateId\":\"95799ea1-aab1-4b20-aadc-d7cfff6fd4c3\"}",
    "metadata": "{\"$correlationId\":\"fbacdd2b-88a3-4b5f-995b-131ff187f660\"}"
  }
]
```

The following:
```
"metadata": "{\"$correlationId\":\"fbacdd2b-88a3-4b5f-995b-131ff187f660\"}"
```
is valid JSON but the type of "metadata" will be a **string** and not a **JSON object** (unless it is parsed again a second time but there is no point in doing this if we can use a pure JSON object in the first place)

The $by_correlation_id projection currently only accepts pure JSON metadata.

From your last reply on the EventStore google group, you mentioned that the following worked:
    "metadata": {"$correlationId":"fbacdd2b-88a3-4b5f-995b-131ff187f660"}

Can you please confirm if the issue is resolved or is there something I'm missing?

**Comment by shaan1337** [comment]
@krofka we have fixed this issue through PR #1881 and #1882. I was missing the fact that EventStore already accepted escaped JSON strings as data/metadata. thanks

**Comment by krofka** [comment]
@shaan1337 Thanks
